asf_heat_pump_affordability/pipeline/produce_costs_dataset.py

```diff
@@ -114,20 +114,20 @@ def main(
     # Get archetypes
     archetypes_dict = archetypes.classify_dict_archetypes_masks(mcs_epc_inf)
 
-    # Get cost percentiles
-    if not cost_percentiles:
```

Comment:
NB This works if `cost_percentiles` is a list, because `bool([1,2,3])` evaluates to `True`, however it doesn't work if `cost_percentiles` is a numpy array (which you permit in the type hint), so `bool(numpy.array([1,2,3]))` returns a ValueError (The truth value of an array with more than one element is ambiguous. Use a.any() or a.all())

As such in this case, you most likely want to evaluate as: `if cost_percentiles is not None:`